client/ayon_core/lib/transcoding.py

```diff
@@ -834,6 +834,112 @@ def get_ffprobe_streams(path_to_file, logger=None):
     return get_ffprobe_data(path_to_file, logger)["streams"]
 
 
+def get_image_info_metadata(
+    path_to_file,
+    keys=None,
+    logger=None
+):
+    """Get metadata from image file.
+
+    At first it will try to detect if the image input is supported by
+    OpenImageIO. If it is then it gets the metadata from the image using
+    OpenImageIO. If it is not supported by OpenImageIO then it will try to
+    get the metadata using FFprobe.
+
+    Args:
+        path_to_file (str): Path to image file.
+        keys (list[str]): List of keys that should be returned. If None then
+            all keys are returned. Keys are expected all lowercase.
+            Additional keys are:
+            - "framerate" - will be created from "r_frame_rate" or
+                "framespersecond" and evaluated to float value.
+        logger (logging.Logger): Logger used for logging.
+    """
+    if logger is None:
+        logger = logging.getLogger(__name__)
+
+    def _ffprobe_metadata_conversion(metadata):
+        """Convert ffprobe metadata unified format."""
+        output = {}
+        for key, val in metadata.items():
+            if key in ("tags", "disposition"):
+                output.update(val)
+            else:
+                output[key] = val
+        return output
+
+    metadata_stream = None
+    ext = os.path.splitext(path_to_file)[-1].lower()
+    if ext not in IMAGE_EXTENSIONS:
+        logger.info((
+            "File extension \"{}\" is not supported by OpenImageIO."
+            " Trying to get metadata using FFprobe."
+        ).format(ext))
+        ffprobe_stream = get_ffprobe_data(path_to_file, logger)
+        if "streams" in ffprobe_stream and len(ffprobe_stream["streams"]) > 0:
+            metadata_stream = _ffprobe_metadata_conversion(
+                ffprobe_stream["streams"][0])
```

Comment:
What is is first stream audio?

Reply:
If I understand the question correctly - are you asking if I am sure that the first stream will always be video?

Is there chance this can happen that steam 0 will be audio?

Reply:
If input contains both video and audio, then audio is usually before video.